presto-manager-agent/src/main/java/com/teradata/prestomanager/agent/TarController.java

```diff
@@ -0,0 +1,235 @@
+package com.teradata.prestomanager.agent;
+
+import com.google.inject.Inject;
+import io.airlift.log.Logger;
+
+import javax.ws.rs.client.Client;
+
+import java.io.IOException;
+import java.net.URL;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+import static com.google.common.io.MoreFiles.deleteRecursively;
+import static com.teradata.prestomanager.agent.AgentFileUtils.downloadFile;
+import static java.lang.String.format;
+import static java.nio.file.Files.createDirectories;
+import static java.nio.file.Files.delete;
+import static java.nio.file.Files.isDirectory;
+import static java.nio.file.Files.isRegularFile;
+import static java.util.Objects.requireNonNull;
+
+public class TarController
+        extends PackageController
+{
+    private static final Logger LOGGER = Logger.get(RpmController.class);
+
+    private final Path configDir;
+    private final Path catalogDir;
+    private final Path dataDir;
+    private final Path pluginDir;
+    private final CommandExecutor executor;
+    private final PrestoConfigDeployer configUtils;
+
+    /**
+     * This variable should be configurable when supporting
+     * multiple tarball installations within a single machine
+     */
+    private static final Path installationDir = Paths.get("~/usr/presto");
+    private static Path launcherScript;
+
+    @Inject
+    TarController(PrestoConfig config, Client client,
+            PrestoInformer informer, CommandExecutor executor,
+            PrestoConfigDeployer configUtils)
+    {
+        super(client, informer);
+        configDir = requireNonNull(config.getConfigurationDirectory());
```

Comment:
`this` is needed only when there is name conflict right? what is the use of `this` here?